app/models/concerns/orchestration.rb

```diff
@@ -119,6 +134,12 @@ def process(queue_name)
     # if we have no failures - we are done
     return true if q.failed.empty? and q.pending.empty? and q.conflict.empty? and orchestration_errors?
 
+    fail_queue(q)
+
+    rollback
+  end
+
+  def fail_queue(q)
     logger.warn "Rolling back due to a problem: #{q.failed + q.conflict}"
```

Comment:
This is logging an empty list of problems when the rollback is triggered by the `rescue` from around_save, it'd be better to move the logger into the caller.